lib/src/arithmetic/multiplier.dart

```diff
@@ -357,7 +357,15 @@ class CompressionTreeMultiplier extends Multiplier {
               {String name})
           seGen = CompactRectSignExtension.new,
       bool use42Compressors = false,
-      super.name = 'compression_tree_multiplier'}) {
+      super.name = 'compressison_tree_multiplier'})
+      : super(
+            definitionName: 'CompressionTreeMultiplier_W${a.width}x'
+                '${b.width}_'
+                '${signedMultiplicand ? 'SD_' : ''}'
+                '${signedMultiplier ? 'SM_' : ''}'
+                '${selectSignedMultiplicand != null ? 'SSD_' : ''}'
+                '${selectSignedMultiplier != null ? 'SSM_' : ''}'
+                'with${adderGen(a, b).definitionName}') {
```

Comment:
this call to `adderGen` i think will actually construct an adder connected to those ports, which could lead to additional verilog generation (_outside_ of the module, even) and also slow down simulation since it's simulating a whole additional adder.  i think i remember seeing this pattern in a few other places as well and didnt think it through.